test/peer-id.spec.js

```diff
@@ -63,27 +66,68 @@ describe('PeerId', () => {
     }).to.throw(/immutable/)
   })
 
-  it('recreate an Id from Hex string', () => {
+  it('recreate from Hex string', () => {
     const id = PeerId.createFromHexString(testIdHex)
-    expect(testIdBytes).to.deep.equal(id.id)
+    expect(testIdBytes).to.deep.equal(id.toBytes())
   })
 
-  it('Recreate an Id from a Buffer', () => {
+  it('recreate from a Buffer', () => {
     const id = PeerId.createFromBytes(testIdBytes)
     expect(testId.id).to.equal(id.toHexString())
+    expect(testIdBytes).to.deep.equal(id.toBytes())
   })
 
-  it('Recreate a B58 String', () => {
+  it('recreate from a B58 String', () => {
     const id = PeerId.createFromB58String(testIdB58String)
     expect(testIdB58String).to.equal(id.toB58String())
+    expect(testIdBytes).to.deep.equal(id.toBytes())
   })
 
-  it('Recreate from a Public Key', async () => {
+  it('recreate from CID object', () => {
+    const id = PeerId.createFromCID(testIdCID)
+    expect(testIdCIDString).to.equal(id.toString())
+    expect(testIdBytes).to.deep.equal(id.toBytes())
+  })
+
+  it('recreate from Base58 String (CIDv0))', () => {
+    const id = PeerId.createFromCID(testIdB58String)
+    expect(testIdCIDString).to.equal(id.toString())
+    expect(testIdBytes).to.deep.equal(id.toBytes())
+  })
+
+  it('recreate from CIDv1 Base32', () => {
+    const id = PeerId.createFromCID(testIdCIDString)
+    expect(testIdCIDString).to.equal(id.toString())
+    expect(testId.id).to.equal(id.toHexString())
+  })
+
+  it('recreate from CID Buffer', () => {
+    const id = PeerId.createFromCID(testIdCID.buffer)
+    expect(testIdCIDString).to.equal(id.toString())
+    expect(testIdBytes).to.deep.equal(id.toBytes())
+  })
+
+  it('throws on invalid CID value', () => {
+    const invalidCID = 'QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L'
```

Comment:
Could we add a comment on why this is an invalid cid for future reference? The error doesn't make it terribly clear and at first glance it looks like a valid b58 id.

Reply:
The function code is `0x50`, which does not represent a hash function.
(we explicitly whitelist codes in [js-multihash/src/constants.js#L345](https://github.com/multiformats/js-multihash/blob/b85999d5768bf06f1b0f16b926ef2cb6d9c14265/src/constants.js#L345))
Will add this as a comment.